src/base/Notifier.sol

```diff
@@ -30,27 +29,23 @@ abstract contract Notifier is INotifier {
     mapping(uint256 tokenId => ISubscriber subscriber) public subscriber;
 
     modifier onlyIfApproved(address caller, uint256 tokenId) virtual;
-    modifier onlyValidConfig(uint256 tokenId, PositionConfig calldata config) virtual;
-
-    function _positionConfigs(uint256 tokenId) internal view virtual returns (PositionConfigId storage);
+    function _setUnsubscribe(uint256 tokenId) internal virtual;
+    function _setSubscribe(uint256 tokenId) internal virtual;
 
     /// @inheritdoc INotifier
-    function subscribe(uint256 tokenId, PositionConfig calldata config, address newSubscriber, bytes calldata data)
+    function subscribe(uint256 tokenId, address newSubscriber, bytes calldata data)
```

Comment:
i found the flow of setting subscription/unsubscription very hard to follow

so we have 2 ways of showing that someone is subscribed (as previously), both in the new PositionInfo and in the mapping

The mapping `subscriber` is easy to follow
- look it up in the mapping
- check theyre not subscribed already
- set the new subscriber

To check that position info boolean is subscribed
- external function `subscribe`
- calls virtual function `_subscribe`
- implementation of `_subscribe` is in position manager
- that implementation calls another function `setSubscribe`
- when you read `setSubscribe` it says `/// @ dev this does not actually set any storage`... so its still not set that its subscribed
- then back to position manager to find the actual line that then sets the return data into storage

Reply:
basically that second flow feels quite unnatural to follow currently, trying to think through how it could be better designed

Reply:
we used to be able to set it in notifiers but bc the mapping is now a uint256 I cant pass a storage flag so it has to be set in manager.

if you have suggestions lmk!